pkg/api/controller.go

```diff
@@ -41,8 +42,28 @@ type Controller struct {
 func NewController(config *config.Config) *Controller {
 	var controller Controller
 
-	logger := log.NewLogger(config.Log.Level, config.Log.Output)
+	if config != nil && config.Extensions != nil {
```

Comment:
Why not move all this into LoadConfiguration() itself?

Reply:
In that case, we definitely need one place to LoadConfiguration() even for server so that we avoid this mangling. 
I would LoadConfiguration() and mangle and then start various enabled features.